### PR TITLE
add scalar comparison operations and rand_like to autodiff

### PR DESCRIPTION
**Pull request by ngimel**

### CODE REVIEW

File: torch/csrc/jit/autodiff.cpp [hunk 1]
```diff
@@ -389,7 +402,10 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
       }
     } else if (comparison_ops.find(node)) {
       return {nullptr, nullptr};
-
+    
+    } else if (node->matches("aten::rand_like(Tensor self) -> Tensor")) {
+      return {nullptr};
```
Comment (apaszke) [inline torch/csrc/jit/autodiff.cpp]: I’m not 100% sure that this won’t lead us to some pretty weird errors in AD later. I’ll need to check that later today, so please don’t merge yet.

**Comment by ezyang** [comment]
Hi @apaszke, any update on this?

**Review by apaszke — APPROVED** [review]
Hmm it's just like I suspected -- returning `nullptr` can cause us to emit undefined tensors that will escape the JIT graph, but the same bug is already present via `type_as`, and should be fixed independently. LGTM.

**Comment by ngimel** [comment]
Can this go in?

**Comment by zou3519** [comment]
This needs a rebase after #15072 (test/test_jit.py got refactored a lot), I can rebase this

**Comment by ngimel** [comment]
nm, I'll tackle it onto another PR.

**Pull request closed without merging**